src/math/trig.c

```diff
@@ -99,3 +100,88 @@ int32_t sin_fixed(int32_t th_rad_fxp)
 	/*convert Q2.30 to Q1.31 format*/
 	return sat_int32(Q_SHIFT_LEFT((int64_t)th_rad_fxp, 30, 31));
 }
+/* Compute fixed point cordiccosine with table lookup and interpolation
+ *	The cordic cosine algorithm converges, when the angle is in the range
+ *	[-pi/2, pi/2).If an angle is outside of this range, then a multiple of
+ *	pi/2 is added or subtracted from the angle until it is within the range
+ *	[-pi/2,pi/2).Start with the angle in the range [-2*pi, 2*pi) and output
+ *	has range in [-1.0 to 1.0]
+ *	+------------------+-----------------+--------+--------+
+ *	| thRadFxp	   | cdccosth	     |thRadFxp|cdccosth|
+ *	+----+-----+-------+----+----+-------+--------+--------+
+ *	|WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|
+ *	+----+-----+-------+----+----+-------+--------+--------+
+ *	| 32 | 28  |  1	   | 32	| 31 |	 1   | 4.28   | 1.31   |
+ *	+------------------+-----------------+--------+--------+
+ */
+int32_t cos_fixed(int32_t th_rad_fxp)
+{
+	/*cordic_atan2_lookup_table = atan(2.^-(0:N-1)) N = 31 */
+	/*CORDIC Gain is cordic_gain = prod(sqrt(1 + 2.^(-2*(0:31-1)))) */
+	/*Inverse CORDIC Gain,inverse_cordic_gain = 1 / cordic_gain */
+	static const int32_t inpLUT[CORDICCOSINE_TABLE_SIZE] = { 843314857, 497837829,
```

Comment:
This table seems to be the same as in sin_fixed()? If confirmed please make it module global by moving it to beginning of trig.c. And use same CORDIC_xxx_TABLE_SIZE for both.